frontend/src/metabase/lib/query_time.js

```diff
@@ -163,15 +163,15 @@ export function parseFieldBucketing(field, defaultUnit = null) {
 }
 
 export function parseFieldTarget(field) {
+    if (Number.isInteger(field)) return field;
+
     if (Array.isArray(field)) {
-        if (field[0] === "datetime_field") {
-            return field[1];
-        } if (field[0] === "fk->") {
-            return field;
-        } else {
-            console.warn("Unknown field format", field);
-        }
+        if (field[0] === "field-id")       return field[1];
+        if (field[0] === "fk->")           return field[1];
+        if (field[0] === "datetime_field") return parseFieldTarget(field[1]);
     }
+
+    console.warn("Unknown field format", field);
```

Comment:
I suspect this is going to cause a lot of logs for existing cards that use the legacy bare field-id format. Perhaps we should have a separate deprecation warning?